tests/macros/shared/src/main/scala/scala/macros/tests/scaladays/Serialize.scala

```diff
@@ -0,0 +1,44 @@
+package scala.macros.tests
+package scaladays
+
+import scala.macros._
+
+trait Serialize[T] {
+  def apply(x: T): String
+}
+
+object Serialize {
+  def apply[T](fn: T => String) = new Serialize[T] { def apply(x: T) = fn(x) }
+  implicit def int: Serialize[Int] = Serialize { x => x.toString }
+  implicit def string: Serialize[String] = Serialize { x => "\"" + x + "\"" }
+
+  inline implicit def materialize[T]: Serialize[T] = meta {
```

Comment:
We should change syntax for this to be `implicit def materialize[T]: Serialize[T] = macro { ... }`. See the discussion about this here: https://gitter.im/scalamacros/scalamacros?at=595d65f176a757f808deb35e.